### ISSUE TITLE
Something...is off.

### ISSUE THREAD

**Issue by Treeston** [rngchan/sun-moon-rng#12]
Is it possible that pokeball inheritance is backwards in the script?

Here's my scenario:
I started with the following config.txt:
```======================
MALE/GENDERLESS PARENT
======================
HP IV: 28
ATK IV: 0
DEF IV: 11
SPATK IV: 12
SPDEF IV: 7
SPD IV: 30
HELD ITEM: Destiny Knot
ABILITY SLOT: 2
NATURE: Timid
IS DITTO: N

======================
     FEMALE PARENT    
======================
HP IV: 31
ATK IV: 16
DEF IV: 31
SPATK IV: 31
SPDEF IV: 31
SPD IV: 19
HELD ITEM: Everstone
ABILITY SLOT: HA
NATURE: Bold
IS DITTO: N

======================
     CHILD TRAITS     
======================
HP IV INTERVAL: [31, 31]
ATK IV INTERVAL: [0, 0]
DEF IV INTERVAL: [31, 31]
SPATK IV INTERVAL: [0, 31]
SPDEF IV INTERVAL: [31, 31]
SPD IV INTERVAL: [31, 31]
ABILITY SLOT: Anything
NATURE: Anything
GENDER: M
BALL: F
HIDDEN POWER TYPE: Anything
SHINY: Anything

======================
    RNG PARAMETERS    
======================
STATUS 3: 0x05B2F888
STATUS 2: 0x1C28A916
STATUS 1: 0x1878B84C
STATUS 0: 0x588BA2D3
TSV: 1340
ESV: []

======================
     OTHER OPTIONS    
======================
GENDER RATIO: 1-1
MASUDA METHOD: N
SHINY CHARM: N
SAME SPECIES: Y
SHOW THIS MANY RESULTS: 40
```

This gives me a result after 5 rejects:
```
IV Spread: (F, 31) / (M, 0) / (F, 31) / (F, 31) / (F, 31) / (R, 31)
Ability: HA
Nature: BOLD
Gender: M
Ball Inherited: M
Hidden Power Type: DRAGON
PID: 939ea0bbL
ESV: 818
Shiny: N
Seed to hatch: a2455672L 5a33a45fL bc990effL 5a4039aL
Seed after hatching: 862c1fc2L e60f7ce6L dd1cc484L a6042bb8L
Sequence of actions to hatch:
(This one might not work, still working the details)
>Reject 5 eggs
>Accept egg
(This one will work for sure, but is longer)
>Reject 5 eggs
>Accept egg
```
Hatching this gives me something that _looks_ like the listed mon - except the inherited ball is backwards (it inherited the female's ball).

Looking at the hatched mon using PKHeX also shows that its PID is `8DF7BEDB`, not `939ea0bb` as predicted by the script.

**Comment by Treeston** [comment]
Just had the same issue on another hatch with different balls on both parents. Ball was backwards, PID didn't match what the tool predicted.

Everything else matched tool. Might be a pattern?

**Comment by Treeston** [comment]
Just had this issue again on my third hatch, except this time the balls matched. Everything up to and including IVs matches, PID starts to desync.

Starting with this config file:
```
======================
MALE/GENDERLESS PARENT
======================
HP IV: 31
ATK IV: 0
DEF IV: 31
SPATK IV: 31
SPDEF IV: 31
SPD IV: 18
HELD ITEM: Destiny Knot
ABILITY SLOT: HA
NATURE: Bold
IS DITTO: N

======================
     FEMALE PARENT    
======================
HP IV: 31
ATK IV: 4
DEF IV: 31
SPATK IV: 31
SPDEF IV: 31
SPD IV: 31
HELD ITEM: Everstone
ABILITY SLOT: HA
NATURE: Bold
IS DITTO: N

======================
     CHILD TRAITS     
======================
HP IV INTERVAL: [31, 31]
ATK IV INTERVAL: [0, 0]
DEF IV INTERVAL: [31, 31]
SPATK IV INTERVAL: [0, 31]
SPDEF IV INTERVAL: [31, 31]
SPD IV INTERVAL: [31, 31]
ABILITY SLOT: HA
NATURE: Bold
GENDER: Anything
BALL: Anything
HIDDEN POWER TYPE: Anything
SHINY: Anything

======================
    RNG PARAMETERS    
======================
STATUS 3: 0x0E746679
STATUS 2: 0x9634017C
STATUS 1: 0x650C7881
STATUS 0: 0x8EF6F85E
TSV: 1340
ESV: []

======================
     OTHER OPTIONS    
======================
GENDER RATIO: 1-1
MASUDA METHOD: N
SHINY CHARM: N
SAME SPECIES: Y
SHOW THIS MANY RESULTS: 40
```

Aiming for this target:
```
IV Spread: (F, 31) / (M, 0) / (M, 31) / (R, 27) / (M, 31) / (F, 31)
Ability: HA
Nature: BOLD
Gender: F
Ball Inherited: M
Hidden Power Type: DRAGON
PID: 598a5901L
ESV: 8
Shiny: N
Seed to hatch: bd4eda27L b3f450ccL 3beafcc6L 710d02e5L
Seed after hatching: 563589dbL e2028b32L 620ae798L 9743384dL
Sequence of actions to hatch:
(This one might not work, still working the details)
>Reject 3 eggs
>Accept egg
(This one will work for sure, but is longer)
>Reject 3 eggs
>Accept egg
```

Actual PID generated was `D09F4C74`, ESV of 2510.

**Comment by Treeston** [comment]
According to [this thread](https://www.reddit.com/r/pokemonrng/comments/5moxcm/rng_tool_not_showing_pidpsv/) (no sourcing), the initial PID is taken from PIDRNG, not nursery RNG, so we can't influence it. Only MM/Charm re-rolls use the nursery RNG.

This matches my experience, as I don't have a Shiny Charm on the save, and the parents were matching languages.

**Comment by Treeston** [comment]
In addition to this, the line that determines ball inheritance display is off:
`ball = "M" if tinymt.nextStateAsInt(2) <= 50 else "F"`

This will always display `M`. (Line 299)

**Comment by rngchan** [comment]
Nice catch on the ball roll, that was supposed to be a roll for 100, not 2. I'll double check it in the disassembly later and fix it in the script. As for not using both MM and shiny charm, I'm pretty sure the game still uses one roll to determine something (which should be PID, but apparently isn't), since a call to the RNG is made. That said, it really does seem like predicting PID without MM or shiny charm is impossible. From the disassembly it seemed otherwise, ah well.

Thanks for pointing these out, will correct the ball inheritance thing and put a disclaimer on PID when neither MM or charm is being used, as soon as possible.

**Comment by Treeston** [comment]
<3 ty

**Comment by rngchan** [comment]
Alright, should be fixed as of version 0.10.5. Ball inheritance roll now rolls a number 0-99 as expected, and there is a disclaimer on the PID and ESV values if neither MM or Shiny Charm are being used.
